src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -125,6 +125,12 @@ public ILImporter(MethodDesc method, MethodIL methodIL)
             if (!_methodSignature.IsStatic)
                 _thisType = method.OwningType.InstantiateAsOpen();
 
+            // ECMA-335 II.13.3 Methods of value types, P. 164:
+            // ... By contrast, instance and virtual methods of value types shall be coded to expect a
+            // managed pointer(see Partition I) to an unboxed instance of the value type. ...
+            if (_thisType != null && _thisType.IsValueType)
+                _thisType = _thisType.MakeByRefType();
```

Comment:
This may look better in the !IsStatic block above that computes `_thisType`. Something like:

```
if (!_methodSignature.IsStatic) 
{
    var thisType = method.OwningType.InstantiateAsOpen();

    // ECMA-335 ....
    if (thisType.IsValueType)
        thisType = thisType.MakeByRefType();

    _thisType = thisType;
}
```

You can also mark `_thisType` and similar members as `readonly` while you are on it.